test/object-store/sync/flx_sync.cpp

```diff
@@ -4791,6 +4791,159 @@ TEST_CASE("flx: pause and resume bootstrapping at query version 0", "[sync][flx]
+
+    // Client 3 sets property 'any' to List and inserts two integers in the list.
+    harness.load_initial_data([&](SharedRealm realm) {
+        auto table = realm->read_group().get_table("class_TopLevel");
+        auto obj = table->get_object_with_primary_key(Mixed{foo_obj_id});
+        auto col_any = table->get_column_key("any");
+        set_list_and_insert_element(obj, col_any, 1);
+        set_list_and_insert_element(obj, col_any, 2);
+    });
+
+    // While its session is paused, Client 2 sets property 'any' to List and inserts two integers in the list.
+    CppContext c(realm);
+    realm->begin_transaction();
+    auto obj = table->get_object_with_primary_key(Mixed{foo_obj_id});
+    auto col_any = table->get_column_key("any");
+    set_list_and_insert_element(obj, col_any, 3);
+    set_list_and_insert_element(obj, col_any, 4);
+    realm->commit_transaction();
+
+    realm->sync_session()->resume();
+    wait_for_upload(*realm);
+    wait_for_download(*realm);
+    wait_for_advance(*realm);
+
+    // Client 2 ends up with four integers in the list (in the correct order).
+    auto list = obj.get_list_ptr<Mixed>(col_any);
+    CHECK(list->size() == 4);
+    CHECK(list->get(0) == 1);
+    CHECK(list->get(1) == 2);
+    CHECK(list->get(2) == 3);
+    CHECK(list->get(3) == 4);
+}
+
+TEST_CASE("flx: nested collections in mixed", "[sync][flx][baas]") {
+    Schema schema{{"TopLevel",
+                   {{"_id", PropertyType::Int, Property::IsPrimary{true}},
+                    {"queryable_str_field", PropertyType::String | PropertyType::Nullable},
+                    {"any", PropertyType::Mixed | PropertyType::Nullable}}}};
+
+    FLXSyncTestHarness::ServerSchema server_schema{schema, {"queryable_str_field"}};
+    FLXSyncTestHarness harness("flx_collections_in_mixed", server_schema);
+    SyncTestFile config(harness.app()->current_user(), harness.schema(), SyncConfig::FLXSyncEnabled{});
+
+    // Client 1: {_id: 1, any: ["abc", [42]]}
+    harness.load_initial_data([&](SharedRealm realm) {
+        CppContext c(realm);
+        auto obj = Object::create(c, realm, "TopLevel",
+                                  std::any(AnyDict{{"_id", INT64_C(1)}, {"queryable_str_field", "foo"s}}));
+        auto table = realm->read_group().get_table("class_TopLevel");
+        auto col_any = table->get_column_key("any");
+        auto o = obj.get_obj();
+        o.set_collection(col_any, CollectionType::List);
+        auto list = o.get_list_ptr<Mixed>(col_any);
+        list->add("abc");
+        list->insert_collection(1, CollectionType::List);
+        auto list2 = list->get_list(1);
+        list2->add(42);
+    });
+
+    // Client 2 opens the realm and downloads schema and object created by Client 1.
+    // {_id: 1, any: ["abc", [42]]}
+    auto realm = Realm::get_shared_realm(config);
+    auto table = realm->read_group().get_table("class_TopLevel");
+    auto sub_set = realm->get_latest_subscription_set();
```

Comment:
If you want to simplify things here you could do `subscribe_to_all_and_bootstrap(realm)` to get to a known fully bootstrapped state.